challenge_8/python/mjuiuc/solution.py

```diff
@@ -0,0 +1,23 @@
+from listNode import listNode
+
+
+def solution(head):
+	lop = dict()
+	newHead = listNode()
+	temp = head
+	temp2 = newHead
+	# copy the list first
+	while temp != None:
+		temp2.val = temp.val
+		lop[temp2.val] = temp2
+		temp2.next = listNode()
```

Comment:
Does this mean you'll have an empty node at the very end? The end of a list should be null terminated.